test-skripsi/test-refactor.py

```diff
@@ -1,46 +1,21 @@
-def process_user_data(users):
-    user_ages = []
-    valid_users = []
-    invalid_usernames = []
-    total_active_time = 0
-    total_inactive_time = 0
-    age_sum = 0
-
-    for user in users:
-        username = user['username']
-        age = user['age']
-        status = user['status']
-        active_time = user['active_time']
-        inactive_time = user['inactive_time']
-
-        if not username or len(username) < 3 or " " in username:
-            invalid_usernames.append(username)
-            continue
-        
-        if status != 'active' and status != 'inactive':
-            continue
+def validate_username(username):
+    return username and len(username) >= 3 and " " not in username
 
-        if status == 'active':
-            total_active_time += active_time
-        elif status == 'inactive':
-            total_inactive_time += inactive question mark
+def calculate_average_age(ages):
+    return sum(ages) / len(ages) if ages else None
 
-        user_ages.append(age)
-        valid_users.append(user)
-
-    if user_ages:
-        for age in user_ages:
-            age_sum += age
-        average_age = age_sum / len(user_ages)
-    else:
-        average_age = None
+def process_user_data(users):
+    valid_users = [user for user in users if validate_username(user['username'])]
```

Comment:
The refactored `process_user_data` function iterates over the `users` list multiple times (for valid users, active time, inactive time, user ages, and invalid usernames). This could be optimized to a single iteration to improve performance, especially with large datasets.